File: VB/Controls/LogViewerVB.Wpf/Logging/LogDataStore.vb
Imports CommonVB.Wpf
Imports LogViewerVB.Core

Namespace Logging

    Public Class LogDataStore : Inherits Core.LogDataStore

#Region "Methods"

        Public Overrides Sub AddEntry(logModel As LogModel)

            ' Marshall to Main Thread
            'DispatcherHelper
            Execute(Sub() MyBase.AddEntry(logModel))

        End Sub

#End Region

    End Class

End Namespace

File: CSharp/Controls/LogViewer.Wpf/Logging/LogDataStore.cs
using System.Windows.Threading;
using LogViewer.Core;

namespace LogViewer.Wpf.Logging;

public sealed class LogDataStore : Core.LogDataStore
{
    #region Methods

    public override void AddEntry(LogModel logModel)
        => DispatcherHelper.Execute(() => base.AddEntry(logModel));

    #endregion
}